modules/auxiliary/scanner/mysql/mysql_writable_dirs.rb

```diff
@@ -0,0 +1,83 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::MYSQL
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::Scanner
+
+  def initialize
+    super(
+      'Name'           => 'MYSQL Directory Write Test',
+      'Description'    => %Q{
+          Enumerate writeable directories using the MySQL SELECT INTO DUMPFILE feature, for more
+        information see the URL in the references. ***Note: For every writable directory found,
+        a file called test with the text test will be written to the directory.***
+      },
+      'Author'         => [ 'AverageSecurityGuy <stephen[at]averagesecurityguy.info>' ],
+      'References'  => [
+        [ 'URL', 'https://dev.mysql.com/doc/refman/5.7/en/select-into.html' ]
+      ],
+      'License'        => MSF_LICENSE
+    )
+
+    register_options([
+      OptPath.new('DIR_LIST', [ true, "List of directories to test", '' ]),
+      OptString.new('FILE_NAME', [ true, "Name of file to write", Rex::Text.rand_text_alpha(8) ]),
+      OptString.new('TABLE_NAME', [ true, "Name of table to use - Warning, if the table already exists its contents will be corrupted", Rex::Text.rand_text_alpha(8) ]),
+      OptString.new('USERNAME', [ true, 'The username to authenticate as', "root" ])
+    ])
+
+  end
+
+  # This function does not handle any errors, if you use this
+  # make sure you handle the errors yourself
+  def mysql_query_no_handle(sql)
+    res = @mysql_handle.query(sql)
+    res
+  end
+
+  def run_host(ip)
+    print_warning("For every writable directory found, a file called test with the text test will be written to the directory.")
+    vprint_status("Login...")
```

Comment:
why only print on verbose for these status messages? not saying it's wrong, just not what i'd do